src/mapDataManipulation/runEffectFunctions.ts

```diff
@@ -1,34 +1,33 @@
 import { mapDataManipulation } from 'src/mapDataManipulation/mapDataManipulation';
-import { PredefinedEffects } from 'src/mapDataManipulation/const';
+import { Effects } from 'src/mapDataManipulation/const';
 import {
+  isAnyEffectSet,
   prepareRgbMappingArrays,
   changeRgbMappingArraysInterval,
   prepareManipulatePixel,
 } from 'src/mapDataManipulation/mapDataManipulationUtils';
-import {
-  runGainEffectFunction,
-  runGammaEffectFunction,
-} from 'src/mapDataManipulation/predefinedEffectFunctions';
+import { runGainEffectFunction, runGammaEffectFunction } from 'src/mapDataManipulation/effectFunctions';
 
 // The algorithm works with numbers between 0 and 1, so we must:
 // - change the interval of the values from [0, 255] to [0, 1]
 // - change the values according to the algorithm (first manipulation of gain and then gamma)
 // - change the interval of the values from [0, 1] back to [0, 255]
 
-export async function runPredefinedEffectFunctions(
-  originalBlob: Blob,
-  predefinedEffects: PredefinedEffects,
-): Promise<Blob> {
+export async function runEffectFunctions(originalBlob: Blob, effects: Effects): Promise<Blob> {
+  if (!isAnyEffectSet(effects)) {
+    return originalBlob;
+  }
+
```

Comment:
Nitpick: if user uses:
```
  effects: {
    gain: 1.0,
  }
```
then the transformation will still be applied (though it will be an identity operation). Would it make sense to change `isEffectSet` so that it understands the specific effects and returns false if the value is the same as default?

Not that important though, fftr either way.

EDIT: `runGainEffectFunction` could be simplified with this too, so that it would only take the gain number:
```
const DEFAULT_GAIN_VALUE = 1.0;
...

if (isEffectSet(effects.gain, DEFAULT_GAIN_VALUE)) {
  rgbMappingArrays = runGainEffectFunction(rgbMappingArrays, effects.gain);
}
```

But again, it's not that important, fftr.

Reply:
Just a quick explanation: decided to skip that since `runGainEffectFunction` and  other functions check if the param has default value and return original `rgbMappingArrays` if it has.